### PR TITLE
Fix syntetic bold

### LINKED ISSUE #7659: darktable-elegant themes in Windows: font problem in menus
This is a Windows specific issue, it does not happen in Linux.
When selecting darktable-elegant themes, bold font is not rendered correctly in menus. For example

Cattura6

unlike the default darktable theme

Cattura7

so one can't see which menu item is activated. It affects blend mode menus, as well as presets, and maybe others

Investigating, I have found the following:

- from css we have
```
...
menu,
menu *
...
{
    font-family: "Roboto Light", "Roboto",                  /* best case scenario */
                 "Segoe UI Light", "Segoe UI",              /* Windows 10 default */
                 "SF Pro Display Light", "SF Pro Display",  /* Mac OS X default */
                 "Ubuntu Light", "Ubuntu",                  /* Ubuntu default */
                 "Cantarell",                               /* Gnome default */
                 sans-serif;                                /* default default */
}
```

I have the Roboto font family installed, so the selected font is Roboto Light, otherwise it would be Segoe UI Light. 
In bleng_gui.c the menu font is modified with markup: 

```
mi = gtk_menu_item_new_with_label(_("RGB (display)"));
    if(module_blend_cst == DEVELOP_BLEND_CS_RGB_DISPLAY)
    {
      markup = g_markup_printf_escaped("<span weight=\"bold\">%s</span>", _("RGB (display)"));
      gtk_label_set_markup(GTK_LABEL(gtk_bin_get_child(GTK_BIN(mi))), markup);
      g_free(markup);
    }
```

So the system is asked to render Roboto Light (or Segoe UI Light, same story) in bold. That font face does not exist of course, but while in Linux the bold font is syntesized (faked) by the backend, in Windows the markup is simply disregarded and normal font is rendered.

So one possible fix would be to modify the darktable-elegant css like follows:

```
*,
#lib-plugin-ui,
#iop-plugin-ui,
#bauhaus-popup,
#modules-tabs,
#left,
#right,
combobox,
combobox *,
togglebutton,
togglebutton *,
notebook,
notebook *,
notebook tab,
notebook tab *,
table,
table *,
row,
row *,
frame,
frame *,
alignment,
entry,
entry *,
dialog,
dialog *,
colorswatch,
colorswatch *,
stack,
stack *,
scrollbar,
scrollbar *,
eventbox,
eventbox *,
scale,
scale *,
button,
button *,
treeview,
treeview *,
separator,
eventbox,
eventbox *,
box,
box *
{
    font-family: "Roboto Light", "Roboto",                  /* best case scenario */
                 "Segoe UI Light", "Segoe UI",              /* Windows 10 default */
                 "SF Pro Display Light", "SF Pro Display",  /* Mac OS X default */
                 "Ubuntu Light", "Ubuntu",                  /* Ubuntu default */
                 "Cantarell",                               /* Gnome default */
                 sans-serif;                                /* default default */
}

menu,
menu *
{
    font-family: "Roboto",                  /* best case scenario */
                 "Segoe UI",              /* Windows 10 default */
                 "SF Pro Display Light", "SF Pro Display",  /* Mac OS X default */
                 "Ubuntu Light", "Ubuntu",                  /* Ubuntu default */
                 "Cantarell",                               /* Gnome default */
                 sans-serif;                                /* default default */
}
```

forcing menu items to use the regular Roboto (or Segoe UI) font. That way the bold font is rendered correclty (and better as it is not syntesized):

Cattura8

It doesn't look bad to me, but @Nilvus do you see a better way ?

### PR DESCRIPTION
**Pull request by Mark-64**
This will fix #7659 by moving to css the text attributes of the selected item in the "blending options" menu.
It is a draft for testing, as it requires more work.

@Nilvus @johnny-bit  please test it. See my comment in darktable-elegant-darker.css
If you approve it, I will make the same changes in the iop preset menu and in the other darktable-elegant themes

### CODE REVIEW

**Comment by johnny-bit** [comment]
Looks good imo... however it uses set_name, while I'd preffer if it used gtk_style_context_add_class because IMO it makes sense to have class rather than ID in css

**Comment by Mark-64** [comment]
@johnny-bit using style class instead of ID as requested.
Also the bold attribute is not optional, needed in Linux.

**Comment by Mark-64** [comment]
... and now, for misterious reasons, using style class doesn't work in Windows, while it works with ID.
Did I do something wrong ?

**Comment by Mark-64** [comment]
There a problem in the case of the favourite presets menu
Cattura9

In each menu item the text is half bold and half normal.
This is unsolvable I am afraid with the method in this PR.

At this point I am stuck.
@johnny-bit @Nilvus any other idea ?

**Comment by Nilvus** [comment]
@Mark-64: I would leave as it is. This favorite menu is different from other as he lists presets on all modules. The way it is is really good for me. It helps show quickly the module (in bold) and then name preset. And this is list is to quickly select preset of any module. In other menu, the whole bold line is to quickly show active preset. So different use, different rendering.

**Comment by Mark-64** [comment]
that was with my custom css workaround, the way it is rendered in windows without changes is this
Cattura10

**Comment by Nilvus** [comment]
@Mark-64: just checked darktable master on Linux and I have bold only in module in favorites menu. I rarely use it but so with that check, it's the wanted behavior. And as I said, I really think it's better that way. So what you have on favorite menu in Windows (bold only in module name) is just good. I'm building now your PR to test it on Linux and check all is good. But that PR seems good.

**Review by Nilvus — APPROVED** [review]
Just checked and all good on my side. Nothing broken on any menus where we add bold text (at least what I have in mind) and in all themes.

To answer something you said on your first comment here: you don't need to update other themes. Fonts are set for elegant themes in darker one, as you done. Grey, dark and icons related ones just inherit from what you set in darker one. And darker one inherit for all other things set in darktable.css file.

For me, it's no more draft and seems ready for review and to be merged.

**Comment by Nilvus** [comment]
And one more Windows specific issue fixed! All good. I'm coming now for blind cursor text.

**Comment by Mark-64** [comment]
Well, I fixed what I could.
Blending options and iop preset menus now work correctly also in Windows with darktable-elegant themes.
For the favorite presets menu, the same technique does not apply. 
We either change the font, avoiding the Light variants (screenshot 1), but this will affect other OS too, or we do nothing (screenshot 2) which will not render the module name in bold under Windows.
For the moment I did nothing.

One alternative could be to render the module name in normal and the preset name in italic, that would render correctly in all OS.
It could also make some sense, as bold is used for active items, and there is nothing active in that menu, @Nilvus what do you say?

Othervise the PR is ready for review

**Comment by TurboGit** [comment]
> One alternative could be to render the module name in normal and the preset name in italic,

But italic is harder to read and we want to avoid using it in dt interface.

File: src/gui/presets.c [hunk 1]
```diff
@@ -1241,36 +1241,22 @@ static void dt_gui_presets_popup_menu_show_internal(dt_dev_operation_t op, int32
...
        && !memcmp(bl_params, blendop_params, MIN(bl_params_size, sizeof(dt_develop_blend_params_t)))
        && module->enabled == enabled)
     {
       active_preset = cnt;
       writeprotect = sqlite3_column_int(stmt, 2);
-      char *markup;
-      mi = gtk_menu_item_new_with_label("");
-      if(isdefault)
-      {
-        markup = g_markup_printf_escaped("<span weight=\"bold\">%s %s</span>", name, _("(default)"));
```
Comment (TurboGit) [inline src/gui/presets.c:1255]: instead of  `<span weight=...> </span>` have you tried `<b> </b>` just in case it would work on Windows.

Comment (johnny-bit) [reply src/gui/presets.c:1255]: @TurboGit - I already asked that and it desn't work on windows.

**Comment by TurboGit** [comment]
Ok, this seems to work on my side, I'll let @Nilvus give the go.

**Comment by Nilvus** [comment]
> Well, I fixed what I could.
> Blending options and iop preset menus now work correctly also in Windows with darktable-elegant themes.
> For the favorite presets menu, the same technique does not apply.
> We either change the font, avoiding the Light variants (screenshot 1), but this will affect other OS too, or we do nothing (screenshot 2) which will not render the module name in bold under Windows.
> For the moment I did nothing.
> 
> One alternative could be to render the module name in normal and the preset name in italic, that would render correctly in all OS.
> It could also make some sense, as bold is used for active items, and there is nothing active in that menu, @Nilvus what do you say?

I don't understand the problem here. I thought that before your last commit, all were good on Windows. About favorite presets menu, what's the issue now? You reported that there's a problem in favorite menu by posting following screenshot, and what I answer to you is that's not strange but normal. In preset menu, active preset is bold. In favorite menu, this not the same objective and so, module name is bold to quickly find module name then see name preset in normal text. So if you have bold in module name only in favorite menu, it's all good.

> Cattura9

So, to summarize, if you have bold text in Windows on active preset in preset or blending menu of a module and bold text only in module name in favorite menu, in all themes. So it's all good. I've checked and it's all ok on Linux on my side.

Could you confirm what's ok and what's not on your side before merging?

@TurboGit: you set 3.6 milestone but as it is a fix of UI glitch on Windows, I think 3.4.1 would be better.

**Comment by TurboGit** [comment]
@Nilvus : set to 3.4.1 then.

**Comment by Mark-64** [comment]
@Nilvus,  sorry for not being clear. I posted that screenshot to illustrate that in the case of the favorite menu I can't apply the technique to assign a different style class for the active line (that is then set as bold in css), as all lines are `partially` bold.
So the screenshot in Windows is the second. The first is obtained with the custom css trick mentioned in #7659.

@johnny-bit @TurboGit To be precise, both `<span weight=...> </span>` and  `<b> </b>` work in Windows, but only when using a font-family which has a real bold typeface. Sans, Roboto, Segoe UI, those are all fine. That's why there is no issue with the default darktable theme.
The problem comes when the font-family does not have a real bold typeface. In that case bold must be faked (synthesized) by the backend and while Linux does it, windows does not and ignores the markup. In darktable-elegant themes, Roboto Light or Segoe UI Light are used, and that's why there is the problem.

I opened [this](https://gitlab.gnome.org/GNOME/pango/-/issues/525) issue is Pango, but I don't think it will be fixed any time soon.
Because of those Windows specific Gtk issues it's very difficult to have the same aspect in all OS. A Windows specific css theme could help maybe, just a few lines for the workarounds, all the rest would be inherited

**Comment by Nilvus** [comment]
> @Nilvus, sorry for not being clear. I posted that screenshot to illustrate that in the case of the favorite menu I can't apply the technique to assign a different style class for the active line (that is then set as bold in css), as all lines are `partially` bold.
> So the screenshot in Windows is the second. The first is obtained with the custom css trick mentioned in #7659.

Ok, it's clear. And it's normal that you don't have same thing on favorite menus. In preset menus, it's the active one that have bold effect, so by setting a specific active css tag, you deal with the font issue in Windows.

In favorite menu, there's no active preset as it's just a list of presets to quickly select in the menu. So no active preset at all here. The fix should be different by setting a specific CSS tag only in module name (where <span> bold tag was or is for that menu). But as it's in the same line, I don't know if it could be easily possible to do.

> Because of those Windows specific Gtk issues it's very difficult to have the same aspect in all OS. A Windows specific css theme could help maybe, just a few lines for the workarounds, all the rest would be inherited

Quite annoying those specific Windows Gtk issues but finally, one by one, we succeed to deal with them without, at this stage, affect the rendering in other OS. We need to keep our best to continue deal that way.

About a specific Windows theme, I don't really see how we can do that without charging too much theme list as we have different themes now. And now, the only remaining UI glitch we know at this stage is that preset menu, and at this stage, it can't be fix with CSS.

**Comment by Nilvus** [comment]
This PR is good for me as it doesn't break anything in Linux and all related parts are bold as they should. With last commit and in regard of my last comment, is this all good for you @Mark-64 on Windows? If it's ok for you, I think this could be merged.

**Comment by Mark-64** [comment]
It not all good, but the most annoying defect is fixed. The other glitches are minor and I can't do better for now

**Comment by Nilvus** [comment]
> It not all good, but the most annoying defect is fixed. The other glitches are minor and I can't do better for now

Ok. So ready to be merged @TurboGit. And thanks to your for that work @Mark-64

**Comment by TurboGit** [comment]
Looks good on my side too. Thanks, I'm merging that now.
